src/handlers/wyre.js

```diff
@@ -33,21 +33,33 @@ const wyreApi = axios.create({
   },
 });
 
-class WyreAPIException extends Error {
-  constructor(errorID, message) {
-    super(`${errorID}:${message}`);
-    this.name = 'WyreAPIException';
+const WyreExceptionTypes = {
+  CREATE_ORDER: 'WyreCreateOrderException',
+  TRACK_ORDER: 'WyreTrackOrderException',
+  TRACK_TRANSFER: 'WyreTrackTransferException',
+};
+
+class WyreException extends Error {
+  constructor(name, referenceInfo, errorId, message) {
+    const referenceInfoIds = values(referenceInfo);
+    const referenceId = join(referenceInfoIds, ':');
+    super(`${referenceId}:${errorId}:${message}`);
+    this.name = name;
   }
 }
 
-export const requestWyreApplePay = (
+export const getReferenceId = accountAddress => {
+  const addressLength = accountAddress.length;
+  const lowered = toLower(accountAddress);
+  return lowered.substring(addressLength - 12, addressLength);
+};
+
+export const showApplePayRequest = async (
+  referenceInfo,
```

Comment:
should we keep the alphabetical order here?

Reply:
I'd like to keep the first argument to be the referenceInfo (as I am doing this for many functions) regardless of what the other params are named!